bin/confirm_delivery.py

```diff
@@ -1,88 +1,39 @@
-import argparse
-import csv
-import glob
-import logging
 import os
 import sys
-from collections import defaultdict
-from os.path import join, dirname, abspath, relpath
+import csv
+import argparse
+import logging
 import datetime
 import itertools
-
+from os.path import join, dirname, abspath, relpath
 from cached_property import cached_property
-from egcg_core import clarity
-
+from pyclarity_lims.entities import Step, Queue
+from egcg_core import clarity, util
 from egcg_core.app_logging import AppLogger, logging_default as log_cfg
 from egcg_core.config import cfg
 from egcg_core.rest_communication import get_document, patch_entry
-from pyclarity_lims.entities import Step, Queue
 
 sys.path.append(dirname(dirname(abspath(__file__))))
 from config import load_config
 
-file_extensions_to_check = [
-    'fastq.gz',
-    'bam',
-    'g.vcf.gz'
-]
-
+file_extensions_to_check = ['fastq.gz', 'bam', 'g.vcf.gz']
 lims_workflow_name = 'PostSeqLab EG 1.0 WF'
 lims_protocol_name = 'Data Release EG 2.0 PR'
 lims_stage_name = 'Download Confirmation EG 1.0 ST'
 
 
-def parse_aspera_reports(report_csv):
-    all_files = []
-    with open(report_csv) as open_report:
-        # ignore what is before the second blank lines
-        blank_line = 0
-        for l in open_report:
-            if not l.strip():
-                blank_line += 1
-            if blank_line == 2:
-                break
-
-        dict_reader = csv.DictReader(open_report)
-        for file_dict in dict_reader:
-            if file_dict['level'] == '1':
-                all_files.append((
-                    '/'.join(file_dict['file_path'].split('/')[3:]),
-                    file_dict['ssh_user'],
-                    datetime.datetime.strptime(file_dict['stopped_at'], '%Y/%m/%d %H:%M:%S'),  # 2016/09/08 16:30:27
-                    file_dict['bytes_transferred']
-                ))
-    return all_files
-
-
-def check_end_file_name(f):
-    for ext in file_extensions_to_check:
-        if f.endswith(ext):
-            return True
-    return False
-
-
-def list_files_delivered(path):
-    all_files = {}
-    for root, dirs, files in os.walk(path):
-        for f in files:
-            if check_end_file_name(f):
-                all_files[join(root, f)] = os.stat(join(root, f))
-    return all_files
```

Comment:
These two functions have been moved inside DeliveredSample as `_is_checkable` and `_stat_checkable_files`